ear-production-suite-plugins/plugins/binaural_monitoring/src/orientation_osc.cpp

```diff
@@ -19,17 +24,26 @@ void ListenerOrientationOscReceiver::listenForConnections(uint16_t port) {
   oscPort = port;
   isListening = osc.connect(oscPort);
   updateStatusTextForListenAttempt();
-  // osc.connect always returns true, even if it failed to bind.
-  // instead, we constantly try closing and rebinding if we've not received a message yet.
-  startTimer(timerIdPersistentListen, 1000);
+  if(!isListening) {
+    // To auto-reconnect on connection errors, we constantly try rebinding.
+    startTimer(timerIdPersistentListen, 1000);
+  }
 }
 
 void ListenerOrientationOscReceiver::disconnect() {
   stopTimer(timerIdStatusTextReset);
   stopTimer(timerIdPersistentListen);
   osc.disconnect();
   isListening = false;
-  updateStatusText(std::string("OSC Closed."));
+  updateStatusText(std::string(OSC_CLOSED_MSG));
+  if(onInputTypeChange) onInputTypeChange(InputType::None);
+}
+
+void ListenerOrientationOscReceiver::setInverts(bool y, bool p, bool r, bool qW, bool qX, bool qY, bool qZ)
+{
+  invert = {y, p, r, qW, qX, qY, qZ};
```

Comment:
More nitpicking. I'd be tempted to create an Inversions struct at the call site and pass it in rather than having a long parameter list on this function? Doesn't really matter when it's only called from one place.